mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -738,30 +739,40 @@ private void launchVoiceRecognitionIfNeeded() {
     }
 
     private void openPendingNfcPageIfNeeded() {
-        if (mPendingNfcData == null || mConnection == null || !mStarted) {
+        if (mPendingNfcIntent == null || mPendingNfcIntent.getData() == null
+                || mConnection == null || !mStarted) {
             return;
         }
 
-        Log.d(TAG, "NFC Scheme = " + mPendingNfcData.getScheme());
-        Log.d(TAG, "NFC Host = " + mPendingNfcData.getHost());
-        Log.d(TAG, "NFC Path = " + mPendingNfcData.getPath());
-        String nfcItem = mPendingNfcData.getQueryParameter("item");
-        String nfcCommand = mPendingNfcData.getQueryParameter("command");
-
-        // If there is no item parameter it means tag contains only sitemap page url
-        if (TextUtils.isEmpty(nfcItem)) {
-            Log.d(TAG, "This is a sitemap tag without parameters");
-            // Form the new sitemap page url
-            String newPageUrl = String.format(Locale.US,
-                    "rest/sitemaps%s", mPendingNfcData.getPath());
-            mController.openPage(newPageUrl);
-        } else {
-            Log.d(TAG, "Target item = " + nfcItem);
-            String url = String.format(Locale.US, "rest/items/%s", nfcItem);
-            Util.sendItemCommand(mConnection.getAsyncHttpClient(), url, nfcCommand);
-            finish();
+        Uri pendingNfcData = mPendingNfcIntent.getData();
+
+        Log.d(TAG, "NFC Scheme = " + pendingNfcData.getScheme());
+        Log.d(TAG, "NFC Host = " + pendingNfcData.getHost());
+        Log.d(TAG, "NFC Path = " + pendingNfcData.getPath());
+
+        if ("openhabitem".equals(pendingNfcData.getScheme())) {
+            BackgroundTasksManager.enqueueNfcItemUpload(mPendingNfcIntent);
+        } else if ("openhab".equals(pendingNfcData.getScheme())) {
+            String nfcSitemap = pendingNfcData.getPath();
+            String nfcItem = pendingNfcData.getQueryParameter("item");
+            String nfcState = pendingNfcData.getQueryParameter("command");
+
+            if (TextUtils.isEmpty(nfcItem)) {
+                Log.d(TAG, "This is a sitemap tag without parameters");
+                // Form the new sitemap page url
+                String newPageUrl = String.format(Locale.US, "rest/sitemaps%s", nfcSitemap);
+                mController.openPage(newPageUrl);
+            } else {
+                Log.d(TAG, "This is an old tag for item updates (" + nfcItem + ")");
+                String url = String.format(Locale.US, "rest/items/%s", nfcItem);
+                Util.sendItemCommand(mConnection.getAsyncHttpClient(), url, nfcState);
+                new AlertDialog.Builder(this)
+                        .setMessage(getString(R.string.info_write_tag_again, nfcItem, nfcState))
+                        .setPositiveButton(android.R.string.ok, (dialog, which) -> finish())
+                        .show();
```

Comment:
Maybe just show a snackbar instead of this? Another approach: Route all NFC stuff to the transparent activity and just start MainActivity if the tag is for a sitemap page.

Reply:
I'll try your second approach.

Reply:
Done! Works pretty good and we need no deprecation.